flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -525,12 +525,12 @@ public void setInitialState(@Nullable JobManagerTaskRestore taskRestore) {
 						slotProvider.allocateSlot(
```

Comment:
I think callbacks on the `logicalSlotFuture` no longer need to be executed in the main thread executor, because the `logicalSlotFuture` should be completed by the main thread. This should affect line `Execution.java:548`.